### DISCUSSION TITLE
keyboard maestro macro, permission?

### DISCUSSION THREAD

**Discussion by Halem998** [kovidgoyal/kitty#4057] [Q&A]
hi, i've been using a macro for opening my `terminal.app` with keyboard maestro (see screenshot below).
so just yesterday i switched to `kitty`.

this is not really a bug but i wouldn't know how to ask this otherwise!

**the problem:**
i changed the app in my macro to "kitty".
in my dock ( i'm on macOS) the kitty app symbol starts jumping. 
but the little black dot which indicates the program running doesn't appear, 
in short: keyboard maestro cannot open kitty but i can see it tries to.

**so my question is:** 
upon installing kitty, did i miss some admin permission or sth like that?
sorry for the wording, i'm not too tech savvy ...
btw i installed kitty via `brew`...

thx in advance

![Screenshot 2021-09-23 at 13 10 25](https://user-images.githubusercontent.com/69585777/134497227-8c277c77-919c-4f14-9d05-71b5dcb8077e.png)

**Comment by Luflosi** [comment]
Does kitty open as expected when you start it from Finder?

**Comment by Halem998** [comment]
> Does kitty open as expected when you start it from Finder?

yes,sure.

just KM seems to not have access to it ...?

**Comment by Luflosi** [comment]
If kitty already opened, even without any windows, it seems to work fine. If kitty is closed, it seems to start and then immediately quit, I don't know why. Maybe the environment is different somehow. I couldn't find anything helpful Console.app.

**Comment by kovidgoyal** [comment]
On Thu, Sep 23, 2021 at 05:06:30AM -0700, Luflosi wrote:
> If kitty already opened, even without any windows, it seems to work fine. If kitty is closed, it seems to start and then immediately quit, I don't know why. Maybe the environment is different somehow. I couldn't find anything helpful Console.app.

Run it via a script that captures stdout/stderr to check.

**Comment by Luflosi** [comment]
That didn't help because executing kitty via a script worked, even executing kitty.app/Contents/MacOS/kitty directly worked. The "Activate a specific Application" action seems to do something differently than the "Execute shell script" action.
@tjallemaal as a workaround, you can create the action "Execute shell script" with  `/Applications/kitty.app/Contents/MacOS/kitty -1` as the script.

**Reply by Halem998** [reply]
thx! the shell script works just fine

**Reply by Halem998** [reply]
i guess it's not kitty's fault.
it's not the first time KM is bugging me

**Reply by Halem998** [reply]
but the problem with the shell script is, the macro continues to run after being triggered ...
when i halt it, kitty is being shut down?!
do i have to modify the macro further?
otherwise i'll give up :/

![Screenshot 2021-09-23 at 15 48 35](https://user-images.githubusercontent.com/69585777/134518976-eeafca02-c6fd-41c5-9a64-a6b8af8b80f1.png)

**Reply by Luflosi** [reply]
Try changing "display results in a window" to "asynchronously".

**Reply by Halem998** [reply]
you made my day !

**Comment by page-down** [comment]
You may also consider using [hammerspoon](https://github.com/Hammerspoon/hammerspoon) , which is powerful but does not have a graphical interface for configuration.

After installation, click the menu bar icon and select `Open Config`, then put the following code in the `init.lua` configuration file.

```lua
hs.hotkey.bind({'control', 'option', 'command'}, 't', function() hs.application.launchOrFocusByBundleID('net.kovidgoyal.kitty') end)
```

The configuration file is located at `~/.hammerspoon/init.lua` by default.
